addons/mail/static/src/core/common/composer.js

```diff
@@ -498,7 +509,18 @@ export class Composer extends Component {
                 if (!isDiscard && this.props.composer.thread.type === "mailbox") {
                     this.notifySendFromMailbox();
                 }
-                this.clear();
+                if (
+                    args.length === 0 &&
+                    document
+                        .querySelector(".o_mail_composer_form_view #body_0")
+                        .innerText.replace(/^\s*$/gm, "")
+                ) {
+                    this.saveInputToLocalStorage();
+                    this.setInputFromLocalStorage();
+                } else {
+                    this.clear();
+                }
+                browser.localStorage.removeItem(`${this.thread.localId}_fullComposerText`);
```

Comment:
`browser.localStorage.removeItem()` can probably be called only once if moved in `clear()` method.
And it feels natural that clearing the content does clear the save entry in local storage